app/models/piece.rb

```diff
@@ -1,8 +1,21 @@
 class Piece < ActiveRecord::Base
+  belongs_to :user
   belongs_to :game
 
   def is_obstructed?
     return false
   end
 
+#Capture Pieces Methods - Validates if the other piece is valid to capture, moves current piece to new location
+
+  def move_to!(new_x, new_y)
+    capture_piece = Piece.find_by(x_cord: new_x, y_cord: new_y)
+    return if capture_piece && capture_piece.color == self.color
```

Comment:
it’s checking if there’s a piece there and then the color
i think without it my test fails

> On Jun 28, 2016, at 8:48 PM, Aleks Real <notifications@github.com> wrote:
> 
> In app/models/piece.rb <https://github.com/Methodologists/metchess/pull/16#discussion_r68873897>:
> 
> >    belongs_to :game
> >  
> >    def is_obstructed?
> >      return false
> >    end
> >  
> > +#Capture Pieces Methods - Validates if the other piece is valid to capture, moves current piece to new location
> > +
> > +  def move_to!(new_x, new_y)
> > +    capture_piece = Piece.find_by(x_cord: new_x, y_cord: new_y)
> > +    return if capture_piece && capture_piece.color == self.color
> shouldn't it just be return if capture_piece.color == self.color?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/Methodologists/metchess/pull/16/files/92c6fc0741ffb204cf5751a7073e30f6e22ac763#r68873897>, or mute the thread <https://github.com/notifications/unsubscribe/ALOK7BT4sLwBZjXGOIVlg4mpa1rjo3yCks5qQc7tgaJpZM4I-FiT>.
>